sdk/src/main/java/io/opentelemetry/sdk/metrics/MetricData.java

```diff
@@ -34,33 +37,120 @@
-   * @since 0.1.0
+   * <p>Only one type of points are available at any moment for a {@link MetricData}.
+   *
+   * @return the data {@link Point}s for this metric, or {@code null} if this type of points are not
+   *     accepted.
+   * @since 0.3.0
    */
-  public abstract long getStartEpochNanos();
+  @Nullable
+  public abstract Collection<Int64Point> getInt64Points();
 
   /**
-   * Returns the the epoch timestamp in nanos when data were collected, usually it represents the
-   * moment when {@code Instrument.getData()} was called.
+   * Returns the data {@link DoublePoint}s for this metric, or {@code null} if the {@link
+   * Descriptor.Type} of the metric does not generate this type of points.
    *
-   * @return the epoch timestamp in nanos.
-   * @since 0.1.0
+   * <p>Only one type of points are available at any moment for a {@link MetricData}.
+   *
+   * @return the data {@link Point}s for this metric, or {@code null} if this type of points are not
+   *     accepted.
+   * @since 0.3.0
+   */
+  @Nullable
+  public abstract Collection<DoublePoint> getDoublePoints();
+
+  static MetricData createWithInt64Points(
+      Descriptor descriptor, Collection<Int64Point> int64Points) {
+    Descriptor.Type type = Utils.checkNotNull(descriptor, "descriptor").getType();
+    Utils.checkState(
+        type == Descriptor.Type.NON_MONOTONIC_INT64 || type == Descriptor.Type.MONOTONIC_INT64,
+        "Incompatible points type with metric type.");
+    return new AutoValue_MetricData(
+        descriptor, Utils.checkNotNull(int64Points, "longPoints"), null);
+  }
+
+  static MetricData createWithDoublePoints(
+      Descriptor descriptor, Collection<DoublePoint> doublePoints) {
+    Descriptor.Type type = Utils.checkNotNull(descriptor, "descriptor").getType();
+    Utils.checkState(
+        type == Descriptor.Type.NON_MONOTONIC_DOUBLE || type == Descriptor.Type.MONOTONIC_DOUBLE,
+        "Incompatible points type with metric type.");
+    return new AutoValue_MetricData(
+        descriptor, null, Utils.checkNotNull(doublePoints, "doublePoints"));
+  }
+
+  @Immutable
+  abstract static class Point {
+    Point() {}
+
+    /**
+     * Returns the start epoch timestamp in nanos of this {@code Instrument}, usually the time when
+     * the metric was created or an aggregation was enabled.
+     *
+     * @return the start epoch timestamp in nanos.
+     * @since 0.3.0
+     */
+    public abstract long getStartEpochNanos();
+
+    /**
+     * Returns the the epoch timestamp in nanos when data were collected, usually it represents the
+     * moment when {@code Instrument.getData()} was called.
+     *
+     * @return the epoch timestamp in nanos.
+     * @since 0.3.0
+     */
+    public abstract long getEpochNanos();
+  }
+
+  /**
+   * Int64Point is a single data point in a timeseries that describes the time-varying values of a
+   * int64 metric.
+   */
+  @Immutable
+  @AutoValue
+  public abstract static class Int64Point extends Point {
```

Comment:
How about we call this `LongPoint`, as that will be much more clear to most java developers.

Reply:
I would like to make a decision one way or the other. the protocol will use int64 and other languages will use int64 but I do see the confusion for Java devs.

Reply:
My preference would be to call the class `LongPoint` and document the Int64 protocol information in the javadoc.

Reply:
Can I do that in a separate PR?

Reply:
@jkwatson if you agree with that please review/approve this PR.